### DISCUSSION TITLE
能不能提供工具将图数据集转成只有数字的数据集

### DISCUSSION THREAD

**Discussion by fakeache** [apache/incubator-graphar#155] [Q&A] [answered]
一些框架只支持数据集包含数字，是不是能够提供工具将本来是字符串的图数据集转成只包含数字

**Comment by lixueclaire** [comment, marked as answer]
Hi fakeache,

I understand that you're interested in transferring property graphs to simple graphs. GraphAr is actually designed to support property graphs, but it does assign an index to each vertex (vid) and uses that vid to store sources/destinations for edges. These indices can be generated from primary keys using the [IndexGenerator](https://alibaba.github.io/GraphAr/user-guide/spark-lib.html#indexgenerator) feature in our Spark libraries. This is especially useful for datasets that use original ID or primary keys to represent the vertices, which may be in the form of strings as you mentioned.
